### ISSUE TITLE
[@storybook/vue] v6.1.21 and above: dynamic code snippet auto-generation is broken

### ISSUE THREAD

**Issue by gomorizsolt** [storybookjs/storybook#14559]
After upgrading to either the latest 6.1.x or 6.2.x version (6.1.21 & 6.2.7 respectively), the code snippets don't get auto-magically generated based on the template:

![dynamic_story_source_error](https://user-images.githubusercontent.com/30864202/114391579-f6540300-9b97-11eb-8346-b3d0f178c727.png)

Interestingly though, if the version is pinned to 6.1.20 then the warning no longer shows up.

## Reproduce

1. Clone https://github.com/gomorizsolt/intro-storybook-vue-template.
2. Install dependencies: `yarn`.
3. Launch storybook: `yarn storybook`.
4. Select a component, navigate to the "Docs" tab and click on "Show code" (in fact, the error will show up even if you merely select the component).

![default_code_type](https://user-images.githubusercontent.com/30864202/114391704-1a174900-9b98-11eb-83c2-b25732b46cc3.png)

**Comment by shilman** [comment]
This must have to do with the babel transpilation of prettier for IE11 @pocka ... any ideas?

**Comment by pocka** [comment]
I'm not sure... we can revert the commit but it breaks IE compat again 🤷 

Maybe same as this https://github.com/whitespace-se/storybook-addon-html/issues/46? The empty error message is weird but the stack traces are so similar. (some people posted workarounds there, those would help if this issue has the same cause)

**Comment by gomorizsolt** [comment]
Hey @shilman and @pocka, thanks for quickly replying to the issue! I skimmed through the referenced issue and it seems that upgrading `@babel/core` to the latest version (`^7.13.15`) was enough to resolve the issue (at least in the case of the [example repo](https://github.com/gomorizsolt/intro-storybook-vue-template)). Having said that, you may need to adapt transitive dependencies as well and specify their versions via the `resolutions` field:

```json
{
  "resolutions": {
    "@babel/helper-create-class-features-plugin": "^7.12.13",
    "@babel/helper-replace-supers": "^7.12.13",
    "@babel/plugin-transform-classes": "^7.12.13"
  }
}
```

I had to do it in a project I'm working on. For those of who are not aware of it: keep in mind that [there's no out of the box npm equivalent of yarn `resolutions`](https://stackoverflow.com/questions/52416312/npm-equivalent-of-yarn-resolutions) thus if switching to yarn is not a suitable option for you, you could use [npm-force-resolutions](https://github.com/rogeriochaves/npm-force-resolutions) (or maybe [npm-shrinkwrap.json](https://docs.npmjs.com/cli/v7/commands/npm-shrinkwrap)).

@shilman @pocka Feel free to close the issue if there's nothing to add from your side.

**Comment by catrope** [comment]
This appears to be fixed in Storybook v6.2.9, which pulls in a recent enough version of Babel

**Comment by pocka** [comment]
Closing as there is nothing we can do on our side. If you encountered this issue, please upgrade babel packages (greater than or equal to `7.12.13`).
